### PR TITLE
Bug bash - High priority bug fixes

### PR DESCRIPTION
**Pull request by codemonkey800**
## Description

Fixes most of the high priority bugs found during the bug bash.

### Formatting of API reference attributes is hard to parse

- [Airtable](https://airtable.com/shr6MXADGlNNaohN7/tblKHsTqQzLZW70S9/viwhDGIqzDkq0EprV/recUBJPEabMrunBMU)
- Demos:
  - [API Reference Headings](https://napari-org-high-priority-bugs.vercel.app/api/stable/index.html#modules)
  - [API property and method indentation](https://napari-org-high-priority-bugs.vercel.app/api/stable/napari.layers.Layer.html#id18)

image
image

### [Desktop] left-side contents menu can overlap main body

- [Airtable](https://airtable.com/shr6MXADGlNNaohN7/tblKHsTqQzLZW70S9/viwhDGIqzDkq0EprV/recmK51Gk7Qdk18vv)
- [Demo](https://napari-org-high-priority-bugs.vercel.app/api/stable/napari.layers.Shapes.html)

image

### Spacing between content and bottom of page title is squished at approx >=1425

- [Airtable](https://airtable.com/shr6MXADGlNNaohN7/tblKHsTqQzLZW70S9/viwhDGIqzDkq0EprV/recTYGQ8LFc7HLiRq)
- [Demo](https://napari-org-high-priority-bugs.vercel.app/api/stable/napari.layers.Image.html)

image

### API reference attributes/methods do not show an anchor link

- [Airtable](https://airtable.com/shr6MXADGlNNaohN7/tblKHsTqQzLZW70S9/viwhDGIqzDkq0EprV/recOw2NwpY4tQ8Lmb)
- [Demo](https://napari-org-high-priority-bugs.vercel.app/api/stable/napari.layers.Image.html#napari.layers.Image.affine)

header-link

### CODE REVIEW

File: theme/src/components/GlobalTableOfContents.tsx [hunk 1]
```diff
@@ -25,8 +25,53 @@ const HEADER_TITLES = [Header.Level2, Header.Level3];
...
+      window.getComputedStyle(itemContainerRef.current);
+    const paddingLeft = parseFloat(computedStyle?.paddingLeft ?? '0');
+    const borderLeft = parseFloat(computedStyle?.borderLeft ?? '0');
+
+    for (const [headerId, node] of itemsRef.current.entries()) {
+      // Total item is the node width + the parent container's left border and padding.
+      const width = borderLeft + paddingLeft + node.offsetWidth;
+
+      // Check if current node overflows the parent list node:
+      // https://stackoverflow.com/a/10017343
```
Comment (klai95) [inline theme/src/components/GlobalTableOfContents.tsx:65]: Just want to clarify if this stackoverflow link is purposed to be here for some context or not?

Comment (codemonkey800) [reply theme/src/components/GlobalTableOfContents.tsx:65]: yep this is for context on the approach I'm using 😄

Comment (codemonkey800) [reply theme/src/components/GlobalTableOfContents.tsx:65]: although, it's slightly modified because i'm also including the left border and padding in the total width

**Review by kne42 — APPROVED** [review]
LGTM

**Review by klai95 — APPROVED** [review]
LGTM!

File: theme/src/components/GlobalTableOfContents.tsx (hunk 1 above)
Comment (klai95) [reply theme/src/components/GlobalTableOfContents.tsx:65]: got it!

**Review by liaprins-czi — COMMENTED** [review]
This all looks good to me!
